src/components/Avatar.js

```diff
@@ -62,22 +58,13 @@ const defaultProps = {
 
 function Avatar(props) {
     const [imageError, setImageError] = useState(false);
-    const prevNetworkStatusRef = useRef(props.network.isOffline);
 
-    useEffect(() => {
-        const isReconnecting = prevNetworkStatusRef.current && !props.network.isOffline;
-        if (!imageError || !isReconnecting) {
+    useOnNetworkReconnect(() => {
+        if (!imageError) {
```

Comment:
If we are sticking with custom hook for reconnect then we can just do 
```js
useOnNetworkReconnect(() => {
    setImageError(false);
}
```

Reply:
Yeah I don't really get why this optimization existed before really. It is the reconnect we care about. No update should happen I think if the value is the same?

From React docs...
>If the new value you provide is identical to the current state, as determined by an [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will skip re-rendering the component and its children.